src/commands/plugins.rs

```diff
@@ -469,13 +471,78 @@ async fn list_catalogue_and_installed_plugins() -> Result<Vec<PluginDescriptor>>
     Ok(merge_plugin_lists(catalogue, installed))
 }
 
+fn summarise(all_plugins: Vec<PluginDescriptor>) -> Vec<PluginDescriptor> {
+    use itertools::Itertools;
+
+    let names_to_versions = all_plugins
+        .into_iter()
+        .into_group_map_by(|pd| pd.name.clone());
+    names_to_versions
+        .into_values()
+        .flat_map(|versions| {
+            let (latest, rest) = latest(versions);
+            let Some(mut latest) = latest else {
+                // We can't parse things well enough to summarise: return all versions.
+                return rest;
+            };
+            if latest.installed {
+                // The installed is the latest: return it.
+                return vec![latest];
+            }
+
+            let installed = rest.into_iter().find(|pd| pd.installed);
+            let Some(installed) = installed else {
+                // No installed version: return the latest.
+                return vec![latest];
+            };
+
+            // If we get here then there is an installed version which is not the latest.
+            // Mark the latest as installed (representing, in this case, that the plugin
+            // is installed, even though this version isn't), and record what version _is_
+            // installed.
+            latest.installed = true;
+            latest.installed_version = Some(installed.version);
+            vec![latest]
+        })
+        .collect()
+}
+
+fn latest(mut plugins: Vec<PluginDescriptor>) -> (Option<PluginDescriptor>, Vec<PluginDescriptor>) {
+    let Ok(versions) = plugins
+        .iter()
+        .map(|pd| semver::Version::parse(&pd.version))
+        .collect::<Result<Vec<_>, _>>()
+    else {
+        return (None, plugins);
+    };
+    let Some(latest_ver) = versions.iter().max() else {
```

Comment:
Would it be possible to do `versions.iter().enumerate().max()` to get the index of the max item. This index should match the index of the of the plugin in the plugins collection which means you can avoid turning the version into a string and iterating over the plugins collection again.